@here/harp-mapview/lib/MapView.ts

```diff
@@ -2777,6 +2777,30 @@ export class MapView extends EventDispatcher {
         return new THREE.Vector3((x / width) * 2 - 1, -((y / height) * 2) + 1, 0);
     }
 
+    /**
+     * Get canvas client size in css/client pixels.
+     *
+     * Supports canvases not attached to DOM, which have 0 as `clientWidth` and `clientHeight` by
+     * calculating it from actual canvas size and current pixel ratio.
+     */
+    getCanvasClientSize(): { width: number; height: number } {
+        const { clientWidth, clientHeight } = this.canvas;
+        if (
+            clientWidth === 0 ||
+            clientHeight === 0 ||
+            typeof clientWidth !== "number" ||
+            typeof clientHeight !== "number"
+        ) {
+            const pixelRatio = this.m_renderer.getPixelRatio();
+            return {
+                width: Math.round(this.canvas.width / pixelRatio),
+                height: Math.round(this.canvas.height / pixelRatio)
+            };
+        } else {
+            return { width: clientWidth, height: clientHeight };
+        }
+    }
+
```

Comment:
No need to make it public if the pixel to NDC conversion is done in ScreenProjector.

Reply:
Missed that, fixing it.